### ISSUE TITLE
Other users not notified when a new, unverified, device is added to an account

### ISSUE THREAD

**Issue by cajuntechie** [wireapp/wire-desktop#347]
Wire Version: 2.11.2673 (Desktop)
Operating System: Windows 10

ISSUE DESCRIPTION:

When User A and User B have verified their encryption keys and User A's keys change due to him adding a new device to their account (which is unverified), User B is never notified of the change even when communicating with User A on the new device.

NARRATIVE:

The situation described above presents a simple attack surface for an attacker to MITM Wire users and should be corrected immediately. Basically, anyone with a users password can add a new device to that users account and begin communicating with their contact without their contacts being notified that they are communicating with a new, unverified, device.

**Comment by raphaelrobert** [comment]
It is important in this case that user B verifies all devices of the conversation with A, including his own ones.
Once that is done, a blue shield symbol should appear next to the conversation name on the top.
When A adds a new device, B will not see this in real-time, but as soon as A sends a message, the blue shield will disappear. If B replies to that message, a modal will show up to warn B about the fact that A has added a new device and ask whether B wants to send the message regardless.
Conversely, if B is the first one to send a message after A added a new device, the same modal will show right away.

The mobile clients are more verbose in that respect and show further information inside the conversation when a new device is added. This will also be added to the web app/desktop.

**Comment by gregor** [comment]
We have just encountered a case where the user can never verify all of his own devices due to the local device being added to the database twice. @cajuntechie Did the conversation ever show as verified with the blue shield showing next to the conversation name?

**Comment by DivineOmega** [comment]
@gregor I know someone who has experienced the bug you mentioned here (duplication of local devices). Is there a separate issue for this that I can track?

**Comment by gregor** [comment]
@DivineOmega We do not have a separate public issue tracking this but be assured that we are fixing this as quickly as possble. I opened a PR today https://github.com/wireapp/wire-webapp/pull/679 that you can keep an eye on.

**Comment by DivineOmega** [comment]
@gregor Many thanks.

**Comment by Strit** [comment]
I still think the entire conversation needs to be notified when a user gets on a new device. It just seems more secure, instead of having a "small blue shield" disappear, shich many will not notice.

**Comment by herrmannplatz** [comment]
@Strit this was addressed in #680. already merged and will be part of the next release

**Comment by Strit** [comment]
Great. Thanks.

**Comment by gregor** [comment]
As this has been live for some time, I am going ahead to close this issue.
